variants/BLUE_F407VE_Mini/variant.h

```diff
@@ -0,0 +1,245 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************
+ */
+
+#ifndef _VARIANT_ARDUINO_STM32_
+#define _VARIANT_ARDUINO_STM32_
+
+/*----------------------------------------------------------------------------
+ *        Headers
+ *----------------------------------------------------------------------------*/
+#include "PeripheralPins.h"
+
+#ifdef __cplusplus
+extern "C"{
+#endif // __cplusplus
+
+/*----------------------------------------------------------------------------
+ *        Pins
+ *----------------------------------------------------------------------------*/
+extern const PinName digitalPin[];
+
+enum {
+//External (left to right) 
+  //GND   
+  //VBAT  
+  //5V    
+  PE0,   //D0
+  PE2,   
+  PE4,   
+  PE6,   
+  PC14,  //OSC32_IN
+  PC0,   
+  PC2,   
+  PA0,   
+  PA1,   
+  PA4,   
+  PA6,   //D10
+  PC4,   
+  PB0,   
+  PB2,   
+  PE8,   
+  PE9, 
+  PE11,  
+  PE13,  
+  PE15,  
+  PB11,  
+  PB13,  //D20
+  PB15,
+  PD9,   
+  PD11,  
+  PD13,  
+  PD15,  
+  PC6,   
+  PC8,   
+  PC9, 
+  PA9,   
+  PA11,  //D30 - USB_DM
+  PA13,
+  PA15,
+  PC11,  
+  PC12,  
+  PD1,   
+  PD3,   
+  PD5,   
+  PD7,   
+  PB4,   
+  PB6,   //D40
+  PB8,   
+  //GND   
+  //3V3   
+  //GND   
+//Internal (left to right) 
+  //GND  
+  //3V3  
+  //5V  
+  PE1,
+  PE3,
+  PE5, 
+  PC13,
+  PC15,  //OSC32_OUT
+  PC1, 
+  PC3, 
+  PA2, 
+  PA3,   //D50
+  PA5, 
+  PA7,
+  PC5, 
+  PB1, 
+  PE7, 
+  PE10,
+  PE12,
+  PE14,
+  PB10,
+  PB12,  //D60
+  PB14,
+  PD8, 
+  PD10,
+  PD12,
+  PD14,
+  PC7, 
+  PA8, 
+  PA10,
+  PA12,  //USB_DP
+  PA14,  //D70
+  PC10,
+  PD0, 
+  PD2, 
+  PD4, 
+  PD6, 
+  PB3,  
+  PB5,  
+  PB7,  
+  PB9,   //D79 - LED
+  //GND  
+  //3V3  
+  //GND  
+  // Analog pins
+  PA0_A,   //D80
+  PA1_A,
+  PA2_A,
+  PA3_A,
+  PA4_A,
+  PA5_A,
+  PB0_A,
+  PB1_A,
+  PC0_A,
+  PC1_A,
+  PC2_A,   //D90
+  PC3_A,
+  PC4_A,
+  PC5_A,
+  PEND
+};
+
+// This must be a literal with the same value as PEND
+#define NUM_DIGITAL_PINS        94
+// This must be a literal with a value less than or equal to MAX_ANALOG_INPUTS
+#define NUM_ANALOG_INPUTS       14
+#define NUM_ANALOG_FIRST        80
+
+// Below ADC, DAC and PWM definitions already done in the core
+// Could be redefined here if needed
+// ADC resolution is 12bits
+//#define ADC_RESOLUTION          12
+//#define DACC_RESOLUTION         12
+
+// PWM resolution
+//#define PWM_RESOLUTION          8
+//#define PWM_FREQUENCY           1000
+//#define PWM_MAX_DUTY_CYCLE      255
+
+// On-board LED pin number
+#define LED_BUILTIN             PB9
+
+// On-board user button
+//#define BTN_K_UP                PA0   - no user button
+
+// Below SPI and I2C definitions already done in the core
+// Could be redefined here if differs from the default one
+// SPI Definitions
+#define PIN_SPI_SS              PB7
```

Comment:
Maybe it should be fine to not define SS as PB7 by default as this is the default SDA ? 
![image](https://user-images.githubusercontent.com/20641798/45924605-8364dd00-bf04-11e8-91ea-90a2ecd74ce5.png)
Maybe using PD7

Reply:
What about mapping SPI1 to PA4-7 instead?

Reply:
SS is for Software Select this is not the Hardware one so SS could be map on any pin.
Anyway in that case this could be fine.